src/vlog/vlog-parse.c

```diff
@@ -6474,6 +6474,44 @@ static vlog_node_t p_udp_declaration(void)
    return udp;
 }
 
+static vlog_node_t p_package_declaration(void)
+{
+   // { attribute_instance } package [ lifetime ] package_identifier ;
+   //    [ timeunits_declaration ] { { attribute_instance } package_item }
+   // endpackage [ : package_identifier ]
+
+   BEGIN("package declaration");
+
+   consume(tPACKAGE);
+
+   if (scan(tSTATIC, tAUTOMATIC))
+      p_lifetime();
+
+   vlog_node_t v = vlog_new(V_PACKAGE);
+   ident_t name = p_identifier();
+   vlog_set_ident(v, name);
+   consume(tSEMI);
+
+   while (scan(tWIRE, tUWIRE, tSUPPLY0, tSUPPLY1, tTRI, tTRI0, tTRI1, tTRIAND,
+               tTRIOR, tTRIREG, tWAND, tWOR, tINTERCONNECT, tREG, tSTRUCT,
+               tUNION, tTYPEDEF, tENUM, tSVINT, tINTEGER, tSVREAL, tSHORTREAL,
+               tREALTIME, tTIME, tEVENT, tID, tVAR, tLOGIC, tBIT, tSHORTINT,
+               tLONGINT, tBYTE, tTASK, tFUNCTION, tLOCALPARAM, tPARAMETER))
+      p_package_or_generate_item_declaration(v);
+
+   consume(tENDPACKAGE);
+
+   if (optional(tCOLON)) {
+      ident_t end_name = p_identifier();
+      if (name != end_name)
+         error_at(&state.last_loc, "Trailing identifier '%s' "
+                  "must match the package name '%s'",
```

Comment:
```suggestion
         error_at(&state.last_loc, "trailing identifier '%s' "
                  "must match the package name '%s'",
```

Messages always start with a lower case letter.

Or else

```suggestion
         error_at(&state.last_loc, "'%s' does not match package name '%s'",
```

would match the structure of other similar messages.